src/host/main.js

```diff
@@ -65,8 +73,24 @@ export async function hostMain({
     log(HostMessage.ADD_SERVER_DIRECTORY, { url: serverDirectory.baseUrl });
     setServerOptionsInTestPlan(plan, { baseUrl: serverDirectory.baseUrl });
 
-    log(HostMessage.START_AGENT);
-    await agent.start({ referenceBaseUrl: serverDirectory.baseUrl });
+    const timesOption = getTimesOption(options);
+
+    const emitter = new EventEmitter();
```

Comment:
An event emitter seems like more than we need since we're only looking to resolve a single promise at the first occurrence of a single event. Think we can get away with leaking `abortSignal`'s resolver function and invoking it directly?

Reply:
Let me know if my latest commits address this! I was able to remove the emitter and just trigger via a promise resolver.